addons/stock/models/stock_picking.py

```diff
@@ -444,6 +444,7 @@ def _default_picking_type_id(self):
         ('late', 'Late')], compute='_compute_products_availability')
     show_set_qty_button = fields.Boolean(compute='_compute_show_qty_button')
     show_clear_qty_button = fields.Boolean(compute='_compute_show_qty_button')
+    destination_country_id = fields.Many2one(related='partner_id.country_id', store=True)
```

Comment:
For delivery planning, it is common and necessary to be able to group by the customer's country state. Could you please add this field here?